src/core/utils/EventManager.ts

```diff
@@ -92,35 +96,29 @@ export class EventManager {
                 // TODO: keydown should be matched with existing shortcuts. If
                 // it matches an intent shortcut, trigger the corresponding
                 // intent, otherwise do not trigger a 'keydown' intent.
-                if (
-                    payload.ctrlKey &&
-                    !payload.altKey &&
-                    !payload.shiftKey &&
-                    !payload.metaKey &&
-                    payload.key === 'b'
-                ) {
-                    name = 'applyFormat';
-                    payload = { format: 'bold' };
-                } else if (
-                    payload.ctrlKey &&
-                    !payload.altKey &&
-                    !payload.shiftKey &&
-                    !payload.metaKey &&
-                    payload.key === 'i'
-                ) {
-                    name = 'applyFormat';
-                    payload = { format: 'italic' };
-                } else if (
-                    payload.ctrlKey &&
-                    !payload.altKey &&
-                    !payload.shiftKey &&
-                    !payload.metaKey &&
-                    payload.key === 'u'
-                ) {
-                    name = 'applyFormat';
-                    payload = { format: 'underline' };
-                } else {
-                    return;
+
+                // TODO: use payload.code (Digit[0-6]) instead of
+                // payload.key
+                intentInformation = this.keyMap.fromKeypress({
+                    key: payload.key,
+                    modifiers: {
+                        ctrlKey: payload.ctrlKey,
+                        altKey: payload.altKey,
+                        shiftKey: payload.shiftKey,
+                    },
+                });
```

Comment:
Why create a new interface for something that is standard and that obviously forces us to convert it from a given format to another? I suggest changing the argument type of this function in such way that the payload can be given to it directly. So basically:
```
{
    key: string;
    ctrlKey: boolean;
    altKey: boolean;
    shiftKey: boolean;
}
```